tests/queries/0_stateless/02815_join_algorithm_setting.sql

```diff
@@ -94,3 +94,13 @@ SELECT countIf(explain like '%Algorithm: DirectKeyValueJoin%'), countIf(explain
     EXPLAIN actions = 1
     SELECT s FROM (SELECT toUInt64(9911) id) t1 INNER JOIN dict t2 USING (id)
 );
+
+SET join_algorithm = 'grace_hash';
+
+-- Cannot execute the grace hash with OR condition
+SELECT * FROM ( SELECT number AS key, number * 10 AS key2 FROM numbers_mt(10) ) AS t1
+JOIN ( SELECT k AS key, k + 100 AS key2 FROM t2 ) AS t2 ON t1.key = t2.key OR t1.key2 = t2.key2; -- { serverError NOT_IMPLEMENTED }
+
+-- But for CROSS choose `hash` algorithm even though it's not enabled
+SELECT * FROM ( SELECT number AS key, number * 10 AS key2 FROM numbers_mt(10) ) AS t1
```

Comment:
This query fails with exception:
```
Can't execute any of specified join algorithms for this strictness/kind and right storage type. (NOT_IMPLEMENTED)
```

Reply:
For old analyzer behavior remains, added setting to the test